src/security.ts

```diff
@@ -158,3 +158,80 @@ export class PreviewSecuritySelector {
     this.webviewManager.refresh()
   }
 }
+
+export class AsciidocParserSecurityPolicyArbiter {
+  private readonly enableAsciidocExtensionScriptKey = 'enable_asciidoc_extension_script:'
+
+  constructor (
+    private readonly context: vscode.ExtensionContext
+  ) {
+    this.context = context
+  }
+
+  public getEnableScripts (): boolean {
+    return this.context.workspaceState.get<boolean>(this.enableAsciidocExtensionScriptKey, false)
+  }
+
+  public async setEnableScripts (enabled: boolean): Promise<void> {
+    return this.context.workspaceState.update(this.enableAsciidocExtensionScriptKey, enabled)
+  }
+}
+
+export class AsciidocExtensionSecuritySelector {
+  private static readonly enableAsciidocExtensionScriptKey = 'enable_asciidoc_extension_script:'
+
+  constructor (
+    private readonly apsArbiter: AsciidocParserSecurityPolicyArbiter
+  ) {
+    this.apsArbiter = apsArbiter
+  }
+
+  public async showSelector (): Promise<void> {
+    const enableExtension = this.apsArbiter.getEnableScripts()
+
+    interface ExtensionPickItem extends vscode.QuickPickItem {
+      readonly type: 'disable_extension' | 'enable_extension';
+    }
+
+    function markActiveWhen (when: boolean): string {
+      return when ? '• ' : ''
+    }
+
+    const selection = await vscode.window.showQuickPick<ExtensionPickItem>(
+      [
+        {
+          type: 'disable_extension',
+          label: markActiveWhen(enableExtension === false) + localize('disable_extension.title', 'Disable AsciiDoc extension scripts'),
+          description: localize('disable_extension.description', 'Disable AsciiDoc extension scripts in the workspace.'),
+        }, {
+          type: 'enable_extension',
+          label: markActiveWhen(enableExtension === true) + localize('enable_extension.title', 'Enable AsciiDoc extension scripts'),
+          description: localize('enable_extension.description', 'Enable AsciiDoc extension scripts in the workspace.'),
+        },
+      ], {
+        placeHolder: localize(
+          'asciidocExtensionSecuritySelector.title',
+          'Select security settings for Asciidoc extension scripts in this workspace'),
+      })
+
+    if (!selection) {
+      return
+    }
+    if (selection.type === 'disable_extension') {
+      await this.apsArbiter.setEnableScripts(false)
+    }
+    if (selection.type === 'enable_extension') {
+      if (!enableExtension) {
+        const confirmYes = await vscode.window.showWarningMessage(
+          'AsciiDoc extension will execute scripts in workspace(.asciidoctor/lib/*.js). Do you trust authors of scripts in workspace?',
+          { modal: true },
+          { title: 'Yes, I trust the authors.', value: true },
+          { title: 'No, I don\'t trust the authors.', value: false })
```

Comment:
The modal looks a bit weird with a "Cancel" button in the middle ("Annuler" in French):

![image](https://user-images.githubusercontent.com/333276/173194172-911492bc-6c36-43ff-93f9-044854d5b538.png)


Is it possible to configure button order?

Reply:
Oh... I saw the order as expected in my PC(Windows11).

![image](https://user-images.githubusercontent.com/19510962/173209127-a787b9ed-ae76-41b6-92e8-f175f056ab87.png)

I could not find a way to configure button order. 
Is your PC Linux?

Reply:
I could reproduce it in gWSL2.

![image](https://user-images.githubusercontent.com/19510962/173213688-8b0ee285-014e-4606-83dd-c0e9d8d5ae56.png)

I am thinking this solution.

Reply:
I could not find a way to configure button order when modal is enabled.
So, I changed it to disable "modal".

Reply:
Oh, so this is OS related... To be honest that's not a big deal, thanks for looking into it!

Reply:
I inquired it for vscode project.
https://github.com/microsoft/vscode/issues/151874#event-6791488086

Working as designed....